tests/test_client.py

```diff
@@ -0,0 +1,144 @@
+import unittest
+from unittest.mock import mock_open, patch, call
+from client.client import VariantAnnotationClient, EnsemblError
+from tests import util
+
+TEST_FILENAME = "variants.txt"
+
+
+TEST_URL = "https://test.com/{variant}"
+
+class TestVariantAnnotationClient(unittest.TestCase):
+    def test_parse_variants(self):
+        # This test string includes whitespace
+        TEST_FILE = '''variant1
+variant2    
+variant3'''
+        m = mock_open(read_data=TEST_FILE)
+        client = VariantAnnotationClient(TEST_URL)
+        with patch('builtins.open', m, create=True):
```

Comment:
I opted to go for the with patch(...) as mock: approach, rather than @patch(...) approach since i think that makes it clearer which mock corresponds to which function. I do acknowledge that it makes the code a little less readable (especially with nested with...s, but I think it's worth it. I did try to limit the scope of the with...s to try and minimize the readability impact.